lib/ansible/modules/cloud/amazon/lightsail.py

```diff
@@ -0,0 +1,405 @@
+            except botocore.exceptions.ClientError as e:
+                time.sleep(3)
+
+    # send reboot
+    if inst is not None:
+        try:
+            client.reboot_instance(instanceName=instance_name)
+        except botocore.exceptions.ClientError as e:
+            if e.response['Error']['Code'] != 'NotFoundException':
+                module.fail_json(msg='Unable to reboot instance {0}, error: {1}'.format(instance_name, e))
+        changed = True
+
+    return (changed, inst)
+
+def startstop_instance(module, client, instance_name, state):
+    """
+    Starts or stops an existing instance
+
+    module: Ansible module object
+    client: authenticated lightsail connection object
+    instance_name: name of instance to start/stop
+    state: Target state ("running" or "stopped")
+
+    Returns a dictionary of instance information
+    about the instance started/stopped
+
+    If the instance was not able to state change,
+    "changed" will be set to False.
+
+    """
+    wait = module.params.get('wait')
+    wait_timeout = int(module.params.get('wait_timeout'))
+    wait_max = time.time() + wait_timeout
+
+    changed = False
+
+    inst = None
+    try:
+        inst = find_instance_info(client, instance_name)
+    except botocore.exceptions.ClientError as e:
+        if e.response['Error']['Code'] != 'NotFoundException':
+            module.fail_json(msg='Error finding instance {0}, error: {1}'.format(instance_name, e))
+
+    # Wait for instance to exit transition state before state change
+    if wait:
+        while wait_max > time.time() and inst is not None and inst['state']['name'] in ('pending', 'stopping'):
+            try:
+                time.sleep(5)
+                inst = find_instance_info(client, instance_name)
+            except botocore.exceptions.ClientError as e:
+                time.sleep(1)
+
+    # Try state change
+    if inst is not None and inst['state']['name'] != state:
+        try:
+            if state == 'running':
+                client.start_instance(instanceName=instance_name)
+            else:
+                client.stop_instance(instanceName=instance_name)
+        except botocore.exceptions.ClientError as e:
+            module.fail_json(msg='Unable to change state for instance {0}, error: {1}'.format(instance_name, e))
+        changed = True
+        # Grab current instance info
+        inst = find_instance_info(client, instance_name)
+
+    return (changed, inst)
+
+def core(module):
+    region, ec2_url, aws_connect_kwargs = get_aws_connection_info(module, boto3=True)
+    if not region:
+        module.fail_json(msg='region must be specified')
+
+    client = None
+    try:
+        client = boto3_conn(module, conn_type='client', resource='lightsail',
+                            region=region, endpoint=ec2_url, **aws_connect_kwargs)
+    except (botocore.exceptions.ClientError, botocore.exceptions.ValidationError) as e:
+        module.fail_json(msg=str(e))
```

Comment:
Please add a message like:

`module.fail_json('Failed while connecting to the lightsail service: %s' % e, exception=traceback.format_exc())`

That way, a prettified traceback can be displayed to the user as well.